tests/reference/c-test_builtin_str-7e198f7.stdout

```diff
@@ -0,0 +1,139 @@
+#include <inttypes.h>
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <lfortran_intrinsics.h>
+
+#define ASSERT(cond)                                                           \
+    {                                                                          \
+        if (!(cond)) {                                                         \
+            printf("%s%s", "ASSERT failed: ", __FILE__);                       \
+            printf("%s%s", "\nfunction ", __func__);                           \
+            printf("%s%d%s", "(), line number ", __LINE__, " at \n");          \
+            printf("%s%s", #cond, "\n");                                       \
+            exit(1);                                                           \
+        }                                                                      \
+    }
+#define ASSERT_MSG(cond, msg)                                                  \
+    {                                                                          \
+        if (!(cond)) {                                                         \
+            printf("%s%s", "ASSERT failed: ", __FILE__);                       \
+            printf("%s%s", "\nfunction ", __func__);                           \
+            printf("%s%d%s", "(), line number ", __LINE__, " at \n");          \
+            printf("%s%s", #cond, "\n");                                       \
+            printf("%s", "ERROR MESSAGE:\n");                                  \
+            printf("%s%s", msg, "\n");                                         \
+            exit(1);                                                           \
+        }                                                                      \
+    }
+
+
+struct dimension_descriptor
+{
+    int32_t lower_bound, length;
+};
+// Forward declarations
+void _lpython_main_program();
+void str_conv_for_variables();
+void test_issue_883();
+void test_str_int();
+void test_str_slice_step();
+
+// Implementations
+void _lpython_main_program()
+{
+    test_str_int();
+    str_conv_for_variables();
+    test_str_slice_step();
+    test_issue_883();
+}
+
+void str_conv_for_variables()
+{
+    bool bool_t;
+    char * str_t;
+    int32_t x;
+    float xx;
+    double yy;
+    x = 123;
+    ASSERT(strcmp("123", _lfortran_int_to_str4(x)) == 0);
+    x = 12345;
+    ASSERT(strcmp("12345", _lfortran_int_to_str4(x)) == 0);
+    x = -12;
+    ASSERT(strcmp("-12", _lfortran_int_to_str4(x)) == 0);
+    x = -121212;
+    ASSERT(strcmp("-121212", _lfortran_int_to_str4(x)) == 0);
+    xx =   1.23222339999999999e+01;
+    ASSERT(strcmp(_lfortran_float_to_str4(xx), "12.322234") == 0);
+    yy =   1.23222339999999999e+01;
```

Comment:
These tests will fail after compilation because of precision issues.

Reply:
Can you make them pass? What’s the blocker which’s failing them?

Reply:
Those are most probably bad tests if they compare floating point directly, as that will fail in general, as you found out. So the test should be improved to use `abs(x-y) < eps` approach.

Reply:
> So the test should be improved to use `abs(x-y) < eps` approach.

Correct. I agree.